R/apply_tfrmt.R

```diff
@@ -194,33 +194,61 @@ remove_empty_layers <- function(x, nlayers = 1){
 #'
 #' @importFrom purrr quietly
 #' @importFrom tidyselect starts_with everything
-#' @importFrom dplyr group_by across tally pull
+#' @importFrom dplyr group_by across summarise n tally pull
 #' @importFrom stringr str_detect
 #' @importFrom tidyr unnest
 pivot_wider_tfrmt <- function(data, tfrmt, mock){
 
   # check if data can be transformed wide w/o list columns
-  num_rec_by_grp <- data %>%
-    group_by(across(-!!tfrmt$values)) %>%
-    tally() %>%
-    pull(n)
-  if (any(num_rec_by_grp>1)){
+  num_rec_by_row <- data %>%
+    group_by(across(c(-!!tfrmt$values, -!!tfrmt$param))) %>%
+    summarise(
+      param_list = list(!!tfrmt$param),
+      n = n()
+    )
+
+  if (any(num_rec_by_row$n>1)){
+
     val_fill <- list("")
+
+    suggested_frmt_structs <- num_rec_by_row %>%
+      ungroup %>%
+      filter(n > 1) %>%
+      select(-c(!!!tfrmt$column)) %>%
+      unique() %>%
+      rowwise() %>%
+      mutate(
+        suggested_frmt_struct = frmt_struct_string(
+          grp = list(!!!tfrmt$group),
+          lbl = !!tfrmt$label,
+          param_vals = .data$param_list
+          )
+      ) %>%
+      pull(.data$suggested_frmt_struct) %>%
+      paste0("- `",.,"`",collapse = "\n")
+
+
+   inform(
+     "Multiple param listed for the same group/label. Following frmt_structures may be missing from the body_plan or the order may need to be changed:",
```

Comment:
This is a bit too wide to read can you put more line breaks in

Reply:
done